src/main/webapp/js/finscholar/scholarshippageview/scholarshippageview.soy

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{namespace example.templates.scholarshippageviews}
+{template .scholarshippage}
+  {@param scholarship: [generalInfo: [scholarshipName: string,  
+                                       schoolsList:string,
+                                       introduction: string,
+                                       URL: uri],
+                        requirements: map<string, string>,
+                        applicationNotes: [isRenewable: bool,
+                                           amountPerYear: string,
+                                           applicationProcess: string,
+                                           numberOfYears: int]
+                        ]}
+  <section class="section--center mdl-grid mdl-grid--no-spacing mdl-shadow--2dp">
+    <div class="mdl-card__supporting-text mdl-grid mdl-grid--no-spacing">
+      <div class="page-content">
+        {call .scholarshipIntroduction}
+          {param generalInfo: $scholarship.generalInfo /}
+        {/call}
+        {call .scholarshipRequirements}
+          {param requirements: $scholarship.requirements /}
+        {/call}
+        {call .scholarshipApplication}
+          {param applicationNotes: $scholarship.applicationNotes /}
+        {/call}
+      </div>
+    </div>
+  </section>
+{/template}
+
+{template .scholarshipIntroduction}
+  {@param generalInfo: [scholarshipName: string,  
+                         schoolsList: string,
+                         introduction: string,
+                         URL: uri]}
```

Comment:
Someone correct me if I'm wrong, but it's also possible to make soy template parameters non-nullable as well if that's intended.

`@param! generalInfo:`

Reply:
Yeah that's true